src/torchcodec/decoders/_core/CudaDevice.cpp

```diff
@@ -256,4 +256,36 @@ void convertAVFrameToDecodedOutputOnCuda(
           << " took: " << duration.count() << "us" << std::endl;
 }
 
+// inspired by https://github.com/FFmpeg/FFmpeg/commit/ad67ea9
+void forceCudaCodec(
+    const torch::Device& device,
+    AVCodecPtr* codec,
+    const AVCodecID& codecId) {
+  if (device.type() != torch::kCUDA) {
+    return;
+  }
+
+  const AVCodec* c;
```

Comment:
Let's move this definition into the while loop condition itself. With the change in semantics, we no longer need to reference the `AVCodec` outside of a single loop iteration, so it's fine (and actually good) that its scope is limited to the while loop only.